lib/datadog/profiling/component.rb

```diff
@@ -166,14 +166,12 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
 
         return false if RUBY_VERSION.start_with?('2.3.', '2.4.', '2.5.')
 
-        if Gem.loaded_specs['mysql2']
+        if Gem.loaded_specs['mysql2'] && !compatible_libmysqlclient_version?(settings)
```

Comment:
Should we encapsulate the `Gem.loaded_specs['mysql2']` inside the `compatible_libmysqlclient_version?(settings)` method?